### PR TITLE
Allow args with script mode

### LINKED ISSUE #1652: Step Script should be compatible with Args
# Expected Behavior

I should be able to specify a step script that takes args, and specify those args to the step (possibly via `stepTemplate`).

# Actual Behavior

If a step specifies `script:` and `args:`, it's rejected as invalid.

# Steps to Reproduce the Problem

```
steps:
- name: example
  image: ubuntu
  script: |
    #!/usr/bin/env bash
    echo passed $# args
    echo first arg is $1
    echo all args are $@
  args: ['first', 'second', third']
```

Should print logs:

```
passed 3 args
first arg is first
all args are first second third
```

`script:` should still be invalid if the step specifies `command:`, since those conflict.

### PR DESCRIPTION
**Pull request by ImJasonH**
This allows step.args to be specified alongside step.script. In this
case, the args are passed to the script, where they can be referenced
and used. See example with bash.

This, along with #1653, allows `stepTemplate.args` to pass through to step scripts.

Fixes #1652

/hold
Waiting for v0.10 unless others think this should make it into v0.9

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [y] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [y] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [y] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

**Comment by vdemeester** [comment]
[v0.9.0](https://github.com/tektoncd/pipeline/releases/tag/v0.9.0) got released, we can go ahead and merge it :angel: 
/hold cancel

File: examples/taskruns/step-script.yaml [hunk 1]
```diff
@@ -75,3 +75,12 @@ spec:
           print('Got: ', v)
           quit()
 
+    # Test that args are allowed and passed to the script as expected.
+    - name: args-allowed
+      image: ubuntu
+      args: ['hello', 'world']
+      script: |
+        #!/usr/bin/env bash
```
Comment (afrittoli) [inline examples/taskruns/step-script.yaml]: NIT: we should start adding `set -e` in our scripts, just to promote best practices 👼

File: docs/tasks.md [hunk 2]
```diff
@@ -151,10 +151,12 @@ or container images that you define:
 #### Step Script
 
 To simplify executing scripts inside a container, a step can specify a `script`.
-If this field is present, the step cannot specify `command` or `args`.
+If this field is present, the step cannot specify `command`.
 
 When specified, a `script` gets invoked as if it were the contents of a file in
-the container. Scripts should start with a
+the container. Any `args` are passed to the script file.
```
Comment (bobcatfish) [inline docs/tasks.md]: _yussssss docs_

**Pull request merged**